File: 01 Introdução/1116 Dividindo X por Y/1116.java
import java.util.Scanner;

class Main {
  public static void main(String args[]) {
    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();

    for (int i = 0; i < n; i++) {
      int num = sc.nextInt();
      int dem = sc.nextInt();

      if (dem == 0) {
        System.out.println("divisao impossivel");
      } else {
        System.out.println((float) num / dem);
      }
    }

    sc.close();
  }
}

// https://www.beecrowd.com.br/judge/pt/problems/view/1116
// Ranking 03986º, Time 0.692, Submiss Date 13/02/2023 15:26:45 (Java 14)

File: 01 Introdução/1116 Dividindo X por Y/1116.py
cases = int(input())

for i in range(0, cases):
    num1, num2 = [int(x) for x in input().split(' ')]
    if num2 != 0:
        print(num1 / num2)
        continue
    print('divisao impossivel')

# https://www.beecrowd.com.br/judge/pt/problems/view/1116
# Ranking 05835º, Time 0.184, Submiss Date 08/03/2023 04:54:03 (Python 3.9)
